### PR TITLE
Use Swift Package Manager

### PR DESCRIPTION
**Pull request by tomlokhorst**
(This PR was previously just for the XcodeEdit update)

- [x] Use SPM for XcodeEdit
- [x] Remove R.swift.Library submodule
- [x] Replace OptionKit submodule with SPM Commander
- [x] Use SPM to build `rswift` itself
- [x] Get bitrise to work with SPM
- [x] Update release scripts

### CODE REVIEW

**Comment by tomlokhorst** [comment]
Since it isn't 2014 anymore, maybe we should move away from git submodules; They're annoying.
I'll update this PR to use Swift Package Manager instead, and throw away all the submodules.

**Comment by tomlokhorst** [comment]
Work complete. R.swift can now be build with SPM.

File: Sources/rswift/Rswift.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Tom Lokhorst on 2017-04-22.
+//  From: https://github.com/mac-cain13/R.swift
+//  License: MIT License
+//
+
+import Foundation
+
+struct Rswift {
+  static let version = "3.9.0"
```
Comment (tomlokhorst) [inline Sources/rswift/Rswift.swift]: Version still needs to be manually set.

File: Sources/RswiftCore/Util/UtilExtensions.swift [hunk 2]
```diff
@@ -19,12 +19,29 @@ extension Array {
...
+
+    return groupedBy
   }
 }
 
+// extension Sequence where Iterator.Element : CustomStringConvertible {
+//   func joined(separator: String) -> String {
+//     return map { $0.description }.joined(separator: separator)
+//   }
+// }
```
Comment (mac-cain13) [inline Sources/RswiftCore/Util/UtilExtensions.swift]: Can we just remove this function?

File: Sources/RswiftCore/SwiftTypes/Module.swift [hunk 3]
```diff
@@ -9,23 +9,23 @@
 
 import Foundation
 
-enum Module: ExpressibleByStringLiteral, CustomStringConvertible, Hashable {
+public enum Module: ExpressibleByStringLiteral, CustomStringConvertible, Hashable {
```
Comment (mac-cain13) [inline Sources/RswiftCore/SwiftTypes/Module.swift]: @tomlokhorst Why do these enum + methods need to change to public?

File: Sources/rswift/Rswift.swift (hunk 1 above)
Comment (mac-cain13) [inline Sources/rswift/Rswift.swift]: Yes, need to update the release script that bumps this version.

File: Sources/RswiftCore/SwiftTypes/Module.swift (hunk 3 above)
Comment (tomlokhorst) [inline Sources/RswiftCore/SwiftTypes/Module.swift]: The `rswift` executable now depends on the module `RswiftCore`. These need to be public to be used from the executable.

File: Sources/RswiftCore/Util/UtilExtensions.swift (hunk 2 above)
Comment (tomlokhorst) [inline Sources/RswiftCore/Util/UtilExtensions.swift]: Yes, that is beter. 😄

**Comment by mac-cain13** [comment]
So, release script works, tests all run correctly with Fastlane and from Xcode on my local machine. Only Bitrise is not happy yet, it gives us a linking error when running the tests from Fastlane.

Fixing Bitrise is the only thing preventing me from merging this atm, will take a look if I can fix Bitrise tomorrow!

**Pull request merged**